tokio/src/runtime/time/mod.rs

```diff
@@ -87,33 +88,8 @@ use std::{num::NonZeroU64, ptr::NonNull};
 pub(crate) struct Driver {
     /// Parker to delegate to.
     park: IoStack,
-}
-
-/// Timer state shared between `Driver`, `Handle`, and `Registration`.
-struct Inner {
-    // The state is split like this so `Handle` can access `is_shutdown` without locking the mutex
-    state: Mutex<InnerState>,
-
-    /// True if the driver is being shutdown.
-    is_shutdown: AtomicBool,
 
-    // When `true`, a call to `park_timeout` should immediately return and time
-    // should not advance. One reason for this to be `true` is if the task
-    // passed to `Runtime::block_on` called `task::yield_now()`.
-    //
-    // While it may look racy, it only has any effect when the clock is paused
-    // and pausing the clock is restricted to a single-threaded runtime.
-    #[cfg(feature = "test-util")]
-    did_wake: AtomicBool,
```

Comment:
Moving these atomics into their own `Arc` is best avoided. That's an allocation per integer, and they may be stored far away from each other hurting cache locality.